freqtrade/edge/__init__.py

```diff
@@ -150,10 +160,24 @@ def calculate(self) -> bool:
 
         return True
 
-    def stake_amount(self, pair: str) -> float:
+    def stake_amount(self, pair: str, free_capital: float, total_capital: float) -> float:
+        if pair not in self._cached_pairs:
+            logger.warning("cannot find %s in calculated pairs, "
+                           "stake_amount of strategy is used instead.", pair)
+            return self.config['stake_amount']
```

Comment:
what's the logic in this? the only stake_amount allowed is unlimited (see `__init__` ).

that however makes the message wrong - as it'll use "all" - which is wrong in 2 regards:
* it's not respecting `'capital_available_percentage'` i set in the configuration, which is dangerous as suddenly i have 100USDT (see docs onto why 100) in one trade edge didn't even find.
* the capital is not defined in the strategy but in the config.

Reply:
spot on !

Reply:
any idea what do to in this case ? we can return `free_capital * capital_available_percentage`. it is the best idea I have so far ...

Reply:
it was not necessary after all. the only variable needed for this calculation is stoploss. all other variables are mandatory in edge. and stoploss fallback is already caught.